### PR TITLE
(DOCSP-33540): React Native: Update Connect to an App Services App with auth hooks

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Note for reviewer: the tests are failing due to a known issue when you run all the tests in this test suite. I ran the new test here many times locally and it is passing.

Jira ticket: https://jira.mongodb.org/browse/DOCSP-33540

- [Connect to an App Services App](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33540/sdk/react-native/app-services/connect-to-app-services-app/): Update examples to use new auth hooks.
- [App Provider Reference](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33540/sdk/react-native/api-reference/app-provider/#useapp--): Update `useApp` example to remove old auth pattern.

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-app-services work, if any

### Release Notes

- **React Native SDK**
  - Atlas App Services/Connect to an App Services App: Update examples to use new auth hooks.
  - API Reference/App Provider: Update `useApp` example to remove old auth pattern.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/react-native/v12/TestApp/src/components/app-services/use-app.test.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const logInButton = screen.queryByTestId('log-in');
+
+  if (logInButton) {
+    await user.press(logInButton);
+  }
+
+  const loggedInUserText = await screen.findByTestId('logged-in-user-id');
+  expect(loggedInUserText).not.toEqual(null);
+
+  const notYetLoggedIn = screen.queryByText('No one is logged in yet.');
```
Comment (krollins-mdb) [inline examples/react-native/v12/TestApp/src/components/app-services/use-app.test.tsx:21]: Nice. I wish we could use the `findBy*` for everything because we can `await` results, but as you probably noticed, you can't test a negative with them. If `findBy*` doesn't find what it wants, it errors the whole test. 😢 

This does mean there could be edge cases where `queryByText` isn't in sync with the rendered UI (because it's not waiting to find UI elements). But this is a great use for it.

File: source/sdk/react-native/app-services/connect-to-app-services-app.txt [hunk 2]
```diff
@@ -30,9 +38,11 @@ Configure the App Client
 To set up your App client, pass the App ID string
 to the ``id`` prop of the ``AppProvider``.
 Wrap any components that need to access the App with the ``AppProvider``.
+In this example, we wrap the :ref:`UserProvider <react-native-user-provider>` 
+in the ``AppProvider`` to proceed to authenticating a user.
```
Comment (krollins-mdb) [inline source/sdk/react-native/app-services/connect-to-app-services-app.txt:42]: Minimalist suggestion:
```suggestion
in the ``AppProvider`` to authenticate a user.
```

**Review by krollins-mdb — APPROVED** [review]
Just one minor suggestion. LGTM! Thank you, Dachary!
